### PR TITLE
allow submit of new requests when stopped timers

### PR DESCRIPTION
**Pull request by HagarMeir**
Signed-off-by: Hagar Meir <hagar.meir@ibm.com>

### CODE REVIEW

**Comment by HagarMeir** [comment]
https://github.com/SmartBFT-Go/consensus/issues/309

File: internal/bft/requestpool.go [hunk 1]
```diff
@@ -128,6 +129,10 @@ func (rp *Pool) Submit(request []byte) error {
 		rp.options.RequestTimeout,
 		func() { rp.onRequestTO(reqCopy, reqInfo) },
 	)
+	if rp.stopped {
+		rp.logger.Debugf("pool stopped, submitting with a stopped timer, request: %s", reqInfo)
+		to.Stop()
+	}
```
Comment (yacovm) [inline internal/bft/requestpool.go:135]: this is executed under the lock, so just check if it's stopped right after acquiring the lock, and return early.

File: internal/bft/requestpool.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ type Pool struct {
 	semaphore      *semaphore.Weighted
 	existMap       map[types.RequestInfo]*list.Element
 	timeoutHandler RequestTimeoutHandler
+	closed         bool
 	stopped        bool
```
Comment (yacovm) [inline internal/bft/requestpool.go:55]: Do we need the stopped now ?

**Review by yacovm — CHANGES_REQUESTED** [review]
see comments

File: internal/bft/requestpool.go (hunk 1 above)
Comment (HagarMeir) [reply internal/bft/requestpool.go:135]: we don't return....we submit the request with a stopped timer...

File: internal/bft/requestpool.go (hunk 2 above)
Comment (HagarMeir) [reply internal/bft/requestpool.go:55]: yes...to know during the submit that the pool is stopped and to stop the new request timer

**Review by yacovm — APPROVED** [review]

**Pull request merged**